### DISCUSSION TITLE
A blank white map is loaded (AWT)

### DISCUSSION THREAD

**Discussion by IndiTheo** [mapsforge/mapsforge#1453] [Q&A] [answered]
I'm new to mapsforge, and I'm trying to display a map from .map file on Desktop.
Lacking a comprehensive tutorial, I assembled some sample code from "docs" and samples.
Instead of displaying a map downloaded from BBBike, I get an empty white-gray map.
Code:

```
public class Quickstart {

    public static void main(String[] args) throws Exception {
        MapView mapView = new MapView();
        MapFile mapFile = new MapFile(new File("palermo.map"));
        mapView.getMapScaleBar().setVisible(true);
        TileCache tileCache = AwtUtil.createTileCache(256,
                    mapView.getModel().frameBufferModel.getOverdrawFactor(), 1024,
                    new File("mapcache"));


        TileRendererLayer rendererLayer = new TileRendererLayer(tileCache, mapFile, mapView.getModel().mapViewPosition, AwtGraphicFactory.INSTANCE);
        rendererLayer.setXmlRenderTheme(InternalRenderTheme.DEFAULT);

        mapView.addLayer(rendererLayer);
        mapView.getModel().mapViewPosition.moveCenter(38.1228886,13.3611315);

        mapView.setVisible(true);
        
        JFrame frame = new JFrame();
        frame.setLayout(new BorderLayout());
        frame.setSize(800,600);
        frame.add(mapView);
        frame.setVisible(true);
    }
}
```

What could go wrong here?
Thank you!

**Comment by devemux86** [comment, marked as answer]
Please check [desktop Samples](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-awt/src/main/java/org/mapsforge/samples/awt/Samples.java), it has a complete example working on desktop.

**Reply by IndiTheo** [reply]
Thank you, I adjusted the sample a little and got it to work.

Anyway, I wonder why such a useful library doesn't have exhaustive documentation.
It's a bit difficult to understand how each component works... Tried to read javadocs, and not every class there has a description
